discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -704,6 +704,9 @@ def _set_hostnames_and_roles(self):
                 continue
 
             node = Cluster.find_matching_node(host, nodes)
+            if not node:
+                node_info = [{n.name: {"ips": n.ips, "macs": n.macs}} for n in nodes]
+                log.warn(f"Failed to find matching node for host with mac address {host.macs()} nodes: {node_info}")
```

Comment:
Next line we're accessing ``node.name``, so maybe we sure replace with ``RuntimeError`` or ``assert``